frontend/src/data/DataManager.ts

```diff
@@ -0,0 +1,256 @@
+import { MetadataMap, Dataset } from "./DatasetTypes";
+import { getUniqueProviders } from "./variableProviders";
+import VariableProvider from "./variables/VariableProvider";
+import { joinOnCols } from "./datasetutils";
+import { DataFrame, IDataFrame } from "data-forge";
+import { MetricQuery, MetricQueryResponse } from "./MetricQuery";
+import { getDataFetcher, getDataManager, getLogger } from "../utils/globals";
+import { useEffect } from "react";
+
+async function onMetadataLoaded(callback: (metadata: MetadataMap) => void) {
+  try {
+    const metadata = await getDataManager().loadMetadata();
+    callback(metadata);
+    // Swallow errors - they are logged in the DataManager
```

Comment:
isn't this the DataManager? not sure if you want to be more specific

Reply:
Moved this to its own file, it was kinda awkward that it was here anyway since this is a React hook but the rest of this file is a data loader/cache that isn't React-specific